src/core/config/misc.h

```diff
@@ -87,6 +87,17 @@
 #define OPENTHREAD_CONFIG_ECDSA_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_DETERMINISTIC_ECDSA_ENABLE
+ *
+ * Define to 1 to generate ECDSA signatures determinsitically
+ * according to RFC 6979 instead of randomly.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_DETERMINISTIC_ECDSA_ENABLE
+#define OPENTHREAD_CONFIG_DETERMINISTIC_ECDSA_ENABLE 0
```

Comment:
I think we should continue to use the deterministic digital signature generation procedure of the ECDSA by default (and set it only for TinyCrypt).

Although the deterministic ECDSA is not explicitly required by the Thread specification, SRP protocol which is part of Thread 1.3 and Matter [recommends](https://datatracker.ietf.org/doc/html/rfc8624#section-3.1) to use [RFC6979](https://www.rfc-editor.org/rfc/rfc6979).

Reply:
I agree with @LuDuda . I think we need it for SRP operation.

Reply:
Thanks for the feedback. I had somehow missed that it was recommended by RFC8624. The recommendation as stated by RFC8624 does not provide an explicit reason as to why it makes that recommendation, but a qualified guess would be to avoid issues with systems running out of entropy or not having quality entropy.

I still think it's worthwhile to have an option to compile without deterministic ECDSA, though, for targets which have qualified entropy sources on-board. It saves a whole include chain of dependencies which aren't otherwise needed in other parts of the codebase. But I agree that the default case should be DetECDSA as per the recommendation. Updating the PR shortly.

Reply:
Updated the PR to default back to the pre-existing behaviour, and with a more accurate title & description. I'm not sure what the convention is on OpenThread PR's, do I resolve the conversation or is that the responsibility of the conversation starter after re-reviewing?

Reply:
@stevew817 , thanks for updating this PR.

I am fine to have a build option to disable use of deterministic DSA. They are recommendations in the RFCs after all. I also agree that the default should enable deterministic procedure by default, as we want do want to help ensure proper behavior on platforms that do not provide a cryptographically secure source of randomness .